### PR TITLE
fix(platform/gitlab): Change MR regex to use word boundary

### LINKED ISSUE #20682: Release Notes replace all occurrences of PR with MR
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.154.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.8.3-ee

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

When running renovate to update our kubernetes-manifest that fetch changelog of the release notes from our self-hosted gitlab repo I noticed that renovate change some part of the changelog

For example we use this sampe release notes
image

Here is the evidence.json of the release notes
```json
{"release":{"id":995,"name":"4.0.0","project":{"id":516,"name":"identity-service","created_at":"2022-10-19T03:59:18.300Z","description":""},"tag_name":"4.0.0","created_at":"2023-03-01T09:12:32.521Z","milestones":[],"description":"Env var additons:\r\n\r\n- TEST_PEER\r\n- TEST_PRIO\r\n- TEST_PREEMPTIBLE\r\n- TEST_PUSH\r\n- PRIORITY_PUSH_PREEMPTIBLE\r\n- PEER_PUSH"}}
```

And when renovate fetch the changelog, it become like this
Screenshot 2023-03-01 at 16 21 23

It seems to change anything that have `_PR` into `_MR`

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Found existing PR (repository=s.adi/kubernetes-manifest-production, branch=renovate/mock-security-renovate-repo-4.x)
DEBUG: Fetching changelog: https://snipped/s.adi/identity-service (1.7.14 -> 4.0.0) (repository=s.adi/kubernetes-manifest-production, branch=renovate/mock-security-renovate-repo-4.x)
 WARN: Error linkifying (repository=s.adi/identity-service, branch=renovate/mock-security-renovate-repo-4.x)
       "err": {
         "message": "Missing `repository` field in `options`",
         "stack": "Error: Missing `repository` field in `options`\n    at Function.github (/opt/buildpack/tools/renovate/34.154.1/node_modules/remark-github/index.js:99:11)\n    at freeze (/opt/buildpack/tools/renovate/34.154.1/node_modules/unified/index.js:125:31)\n    at Function.process (/opt/buildpack/tools/renovate/34.154.1/node_modules/unified/index.js:349:5)\n    at linkify (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/util/markdown.ts:42:6)\n    at releaseNotesResult (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/update/pr/changelog/release-notes.ts:196:42)\n    at getReleaseNotes (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/update/pr/changelog/release-notes.ts:155:24)\n    at addReleaseNotes (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/update/pr/changelog/release-notes.ts:427:24)\n    at Object.getChangeLogJSON (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/update/pr/changelog/source-gitlab.ts:153:9)\n    at getChangeLogJSON (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/update/pr/changelog/index.ts:35:15)\n    at embedChangelog (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/changelog/index.ts:16:21)\n    at /opt/buildpack/tools/renovate/34.154.1/node_modules/p-map/index.js:57:22"
       },
       "baseUrl": "https://snipped/"
 WARN: Error linkifying (repository=s.adi/identity-service, branch=renovate/mock-security-renovate-repo-4.x)
       "err": {
         "message": "Missing `repository` field in `options`",
         "stack": "Error: Missing `repository` field in `options`\n    at Function.github (/opt/buildpack/tools/renovate/34.154.1/node_modules/remark-github/index.js:99:11)\n    at freeze (/opt/buildpack/tools/renovate/34.154.1/node_modules/unified/index.js:125:31)\n    at Function.process (/opt/buildpack/tools/renovate/34.154.1/node_modules/unified/index.js:349:5)\n    at linkify (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/util/markdown.ts:42:6)\n    at releaseNotesResult (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/update/pr/changelog/release-notes.ts:196:42)\n    at getReleaseNotes (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/update/pr/changelog/release-notes.ts:155:24)\n    at addReleaseNotes (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/update/pr/changelog/release-notes.ts:427:24)\n    at Object.getChangeLogJSON (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/update/pr/changelog/source-gitlab.ts:153:9)\n    at getChangeLogJSON (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/update/pr/changelog/index.ts:35:15)\n    at embedChangelog (/opt/buildpack/tools/renovate/34.154.1/node_modules/renovate/lib/workers/repository/changelog/index.ts:16:21)\n    at /opt/buildpack/tools/renovate/34.154.1/node_modules/p-map/index.js:57:22"
       },
       "baseUrl": "https://snipped/"
DEBUG: Processing existing PR (repository=s.adi/kubernetes-manifest-production, branch=renovate/mock-security-renovate-repo-4.x)
DEBUG: PR body changed (repository=s.adi/kubernetes-manifest-production, branch=renovate/mock-security-renovate-repo-4.x)
       "prTitle": "Update mock-security-renovate-repo Docker tag to v4"
 INFO: PR updated (repository=s.adi/kubernetes-manifest-production, branch=renovate/mock-security-renovate-repo-4.x)
       "pr": 28,
       "prTitle": "Update mock-security-renovate-repo Docker tag to v4"
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by EasyNV**
## Changes
Change the regex in gitlab platform from `regEx(/PR/g)` to `regEx(/\bPR\b/g)`

## Context
- Closes https://github.com/renovatebot/renovate/issues/20682

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs a test

File: lib/modules/platform/gitlab/index.ts [hunk 1]
```diff
@@ -694,7 +694,7 @@ export async function mergePr({ id }: MergePRConfig): Promise<boolean> {
 export function massageMarkdown(input: string): string {
```
Comment (HonkingGoose) [inline lib/modules/platform/gitlab/index.ts:694]: This change also breaks the normal GitLab Merge Request vocabulary:

```diff
 modules/platform/gitlab/index › massageMarkdown(input) › returns updated pr body

expect(received).toMatchSnapshot()

Snapshot name: `modules/platform/gitlab/index massageMarkdown(input) returns updated pr body 1`

- Snapshot  - 1
+ Received  + 1

@@ -1,8 +1,8 @@
  https://github.com/foo/bar/issues/5 plus also [a link](https://github.com/foo/bar/issues/5

-   Merge Requests are the best, here are some MRs.
+   Merge Requests are the best, here are some PRs.

    ## Open

  These updates have all been created already. Click a checkbox below to force a retry/rebase of any.
  ↵

    at Object.<anonymous> (lib/modules/platform/gitlab/index.spec.ts:2099:46)
    at runMicrotasks (<anonymous>)
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/modules/platform/gitlab/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/modules/platform/gitlab/index.ts:694]: Good catch, is `PRs` the only example you can think of which we'd want to replace?

Comment (HonkingGoose) [reply lib/modules/platform/gitlab/index.ts:694]: We may want to search/replace these, when Renovate detects the GitLab platform?

- `Pull Request` -> `Merge Request`
- `Pull Requests` -> `Merge Requests`
- And the lowercase variants

Comment (rarkins) [reply lib/modules/platform/gitlab/index.ts:694]: Reminder: we ideally only want this term changed _in our own text_. i.e. words we've created. Changing terms in changelogs is an unfortunately side effect

Comment (EasyNV) [reply lib/modules/platform/gitlab/index.ts:694]: > We may want to search/replace these, when Renovate detects the GitLab platform?
> 
> * `Pull Request` -> `Merge Request`
> * `Pull Requests` -> `Merge Requests`
> * And the lowercase variants

Thanks for the feedback 😄, so for this case should I change the regex [on this line](https://github.com/renovatebot/renovate/blob/35.2.0/lib/modules/platform/gitlab/index.ts#L710)
From
`replace(regEx(/Pull Request/g), 'Merge Request')`
To
`replace(regEx(/[Pp][Uu][Ll][Ll] [Rr][Ee][Qq][Uu][Ee][Ss][Tt]/g), 'Merge Request')`?

And am I allowed to update the current test cases to match the proposed fix? Thanks a lot before

Comment (rarkins) [reply lib/modules/platform/gitlab/index.ts:694]: I don't think we have to worry about "false positives" for the string `Pull Request` like we do for `PR`

**Comment by rarkins** [comment]
Closed due to inactivity

**Pull request closed without merging**